src/models/room.ts

```diff
@@ -914,12 +914,38 @@ export class Room extends ReadReceipt<RoomEmittedEvents, RoomEventHandlerMap> {
         return this.myUserId;
     }
 
+    // gets service members (e.g. helper bots) for exclusion
+    private getFunctionalMembers(): string[] {
+        const mFunctionalMembers = this.currentState.getStateEvents(UNSTABLE_ELEMENT_FUNCTIONAL_USERS.name, "");
+        if (Array.isArray(mFunctionalMembers?.getContent().service_members)) {
+            return mFunctionalMembers!.getContent().service_members;
+        }
+        return [];
+    }
+
+    private getInvitedAndJoinedFunctionalMemberCount(functionalMembers: String[]): number {
+        return functionalMembers.reduce((count, m) => {
+            const membership = this.getMembers().find((member) => member.userId === m)?.membership;
+            if (membership && ["join", "invite"].includes(membership)) {
+                return count + 1;
+            }
+            return count;
+        }, 0);
+    }
+
     public getAvatarFallbackMember(): RoomMember | undefined {
-        const memberCount = this.getInvitedAndJoinedMemberCount();
+        const functionalMembers = this.getFunctionalMembers();
+        const invitedAndJoinedFunctionalMemberCount = this.getInvitedAndJoinedFunctionalMemberCount(functionalMembers);
+
+        // only get avatar if conversation is with a single specific other user
+        const memberCount = this.getInvitedAndJoinedMemberCount() - invitedAndJoinedFunctionalMemberCount;
```

Comment:
wouldn't it be easier to iterate through the list of members once, excluding any functional members, rather than first counting the number of functional members, then counting all members, then subtracting one from the other?

Reply:
I originally chose not to do this, so the parts to rip out to remove the functional members feature would be obvious if need be. Now with the refactoring it became a single char change, so I applied this feedback.